### ISSUE TITLE
SWProxy  Rune can not be obtained and the role of information

### ISSUE THREAD

**Issue by shkvincent** [kakaroto/SWProxy#105]
Perform the following steps
1. Change the phone wifi ip Address: 192.168.1.101 and Proxy Port: 8080
2. Run SWProxy and press Start Proxy Server
3. Run Summoners War

problem:
Until into the game , SWProxy still blank , and did not get any information !

Is gaming company as before have modified some key cause SWProxy No information ? Or is SWProxy an updated version ?

**Comment by msramachandran-ymail** [comment]
There was an update y'day night and SWProxy is not working well with the new version. I am getting "Invalid format" on login. I tried to debug and the very first call to version_info_v2.php itself is failing. Why a pass through would fail is beyond me at this point. But someone can fix this thing.

**Comment by shkvincent** [comment]
Is a game update problems caused you ?? If yes , just like the last reason
suddenly can not use identical !! It should please you ! Hard !! grateful !!

2016-06-13 23:33 GMT+08:00 msramachandran-ymail <notifications@github.com>:

**Comment by msramachandran-ymail** [comment]
What baffles me is that it is not even getting to the point of using the keys. That happens only on gateway_v2.php calls. This time it fails during the version_info_v2.php call which is just a pass through. I first thought com2us has implemented an anti-proxy mechanism. But then I ran SW through squid proxy and it works just fine. I will take a closer look after I get home tonight. Could just be a simple enough fix because I was able to decrypt the data using the older key.

**Comment by Xzandro** [comment]
There already is a fix for that: https://github.com/kakaroto/SWProxy/commit/c9c61aa1f333849d542592b207ab7884213e8257

**Comment by msramachandran-ymail** [comment]
Great news thanks :)

**Comment by thatrandomguyo1** [comment]
@Xzandro Could you by chance explain to a lowly pleb how to use that fix? The "initial fix" I tried and am still having errors but I don't know how to use the unreleased "better fix" you linked to.

**Comment by shkvincent** [comment]
Today I tried again a few times , the result was the same , the program did not get anything.

Also this software can RuneManager.0.1.4.0 Traditional Chinese culture? My English is very headache ~

**Comment by Xzandro** [comment]
@thatrandomguyo1 for now you can use this: https://github.com/lstern/SWProxy-plugins/releases

**Comment by thatrandomguyo1** [comment]
@Xzandro Sadly that's the one I've been trying.

**Comment by Xzandro** [comment]
Global server? There are people who can use that fix just fine and others dont. Strange.

**Comment by thatrandomguyo1** [comment]
@Xzandro Correct global, is this just straight thru or trying to gimmick and change at "downloading..."?

**Comment by Xzandro** [comment]
The only change is the host change, because com2us changed that in the recent patch.

**Comment by thatrandomguyo1** [comment]
Periodically it'll get to bringing your friends and flash connecting to server then give me "invalid format" and hang up on connecting never changing till I restart.

**Comment by Xzandro** [comment]
Huh, strange, summoning @lstern

**Comment by thatrandomguyo1** [comment]
So I updated my iOS and now when it hits "invalid format" the proxy reads "Found Summoners War API request : CheckLoginBlock"

**Comment by msramachandran-ymail** [comment]
I tried debugging and gave up. I think there is something fishy in the proxy code. So I am trying to port the code to use CherryProxy. I tried different other proxy codes and it didn't work except for CherryProxy. I am not porting the code locally to it. Just giving you guys a heads up if you want to try and fix the proxy code.

**Comment by lstern** [comment]
Are you guys using IOS? @msramachandran-ymail  is right about the proxy code and it just reminds me of a [recent fix] (https://github.com/kakaroto/SWProxy/commit/a5eab05908e655db652dd3a8f8a67b3badc06a62#diff-4e41814cb10ce59e95537394d69e015a) that may be causing some problems after the hostname change. I'll play with this CherryProxy as its small and clean. Thanks for the tip

**Comment by shkvincent** [comment]
I try this version : 2.1.5 quick fix
If there is not , then do not know how to do ~~
Thank you for your efforts !! nice to have you !!

**Comment by lstern** [comment]
game is in english ?

**Comment by thatrandomguyo1** [comment]
My game is English, global, iOS 9.3.2

**Comment by msramachandran-ymail** [comment]
@istern Yeah I am using iOS and running the proxy on mac. I managed to run the game over cherry proxy with some minimal hack to the original code (not very elegant). I will try to work on extending the proxy when I get time. Will submit a patch if i can :)

**Comment by thatrandomguyo1** [comment]
@lstern Think that latest release had some error in it, when I try to launch it nothing happens, get the windows thinking circle and no launch.

**Comment by Toosick** [comment]
I am getting the same issues everybody is talking about here. Any quick fixes or works in progress?

**Comment by thatrandomguyo1** [comment]
Not sure what changed or why I randomly downloaded it again to try but the latest release is working currently for me. (seemingly the same release I reported crashing my explorer 3 days ago)

**Comment by shkvincent** [comment]
I now use@ thatrandomguyo1 which provided：https://github.com/lstern/SWProxy-plugins/releases
Rune has Keyihuode with character data
Thank you very much for your efforts !!

**Comment by cesium62** [comment]
I wish the fixed release were available in a mac download.  ;-)

**Comment by Toosick** [comment]
The current release of SWProxy doesn't work but running the current source does work. I made a quick tutorial to run it yourself for the non tech people.

Windows tutorial, this worked in Windows 10 as well

Note: Uninstall Python 3.* if you have it and you don't want to cause any problems with installation
1. Download SWProxy source, extract it somewhere: https://github.com/kakaroto/SWProxy/archive/master.zip
2. Install this: https://www.python.org/downloads/release/python-279/
3. Install this: https://www.microsoft.com/en-us/download/details.aspx?id=44266
4. Put this into your CMD prompt: cd "C:\Python27\Scripts\" && pip install yapsy pycrypto dpkt
5. Right click the SWProxy.py file that was installed in step 1
6. Open with: IDLE
7. Click F5 while in the window with code
8. Success. Should be working as normal. Your own your own from here.
